gematria/experiments/access_pattern_bm/contiguous_matrix_test.cc

```diff
@@ -0,0 +1,91 @@
+// Copyright 2023 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gematria/experiments/access_pattern_bm/contiguous_matrix.h"
+
+#include <iostream>
+
+#include "benchmark/benchmark.h"
+
+namespace gematria {
+namespace {
+
+void BM_FlushContiguousMatrixFromCache(benchmark::State &state) {
+  const std::size_t size = state.range(0);
+
+  // Create a random matrix.
+  auto matrix = CreateRandomContiguousMatrix(size);
+
+  for (auto _ : state) {
+    FlushContiguousMatrixFromCache(matrix, size);
+  }
+}
+
+BENCHMARK(BM_FlushContiguousMatrixFromCache)->Range(1 << 4, 1 << 12);
+
+void BM_ContiguousMatrix_NoFlush(benchmark::State &state) {
+  const std::size_t size = state.range(0);
+
+  // Create a random matrix.
+  auto matrix = CreateRandomContiguousMatrix(size);
+#ifdef BALANCE_FLUSHING_TIME
```

Comment:
Nit+optional: I'd recommend changing the use of `BALANCE_FLUSHING_TIME` to

Somewhere in some header:
```
inline constexpr bool kBalanceFlushingTime =
#ifdef BALANCE_FLUSHING_TIME
    true;
#else
    false;
#endif  // BALANCE_FLUSHING_TIME
```
and then in code use
```
std::unique_ptr<int[]> mock;
if (kBalanceFlushingTime) {
  mock = CreateRandomCongiguousMatrix(size);
}
// ...
if (kBalanceFlushingTime) {
  FLushContiguousMatrixFromCache(mock, size);
}
```

The advantage is that most of the code will be parsed by the compiler in any compile and you'd get warned about errors even in the de-activated parts. But at the same time, since the value of `kBalanceFlushingTime` is known at compile time (it's a `constexpr`), the compiler will be able to remove the disabled part via dead code elimination.